include/wx/event.h

```diff
@@ -1863,31 +1872,92 @@ class WXDLLIMPEXP_CORE wxSetCursorEvent : public wxEvent
 
 private:
     wxDECLARE_DYNAMIC_CLASS_NO_ASSIGN(wxSetCursorEvent);
+};
+
+ // Touch Base Event
+
+class WXDLLIMPEXP_CORE wxTouchEventBase : public wxEvent
+{
+public:
+    wxTouchEventBase(wxWindowID winid = 0, wxEventType type = wxEVT_NULL)
+        : wxEvent(winid, type)
+    {
+    }
+
+    wxTouchEventBase(const wxTouchEventBase& event) : wxEvent(event)
+    {
+        m_pos = event.m_pos;
+    }
+
+    const wxPoint& GetPosition() const { return m_pos; }
+    void SetPosition(const wxPoint& pos) { m_pos = pos; }
+
+protected:
+    wxPoint m_pos;
+
+    wxDECLARE_ABSTRACT_CLASS(wxTouchEventBase);
+
+};
+
+// MultiTouch Event
+
+class wxTouchPointId : public wxItemId<void*>
+{
+public:
+    wxTouchPointId() : wxItemId<void*>() { }
+    wxEXPLICIT wxTouchPointId(void* pItem) : wxItemId<void*>(pItem) { }
```

Comment:
It's ok to use just `explicit` in the new code, we don't support any compilers not supporting this keyword any more.